packages/filesystem/src/browser/location/location-renderer.tsx

```diff
@@ -18,34 +18,87 @@ import URI from '@theia/core/lib/common/uri';
 import { LocationService } from './location-service';
 import { ReactRenderer } from '@theia/core/lib/browser/widgets/react-renderer';
 import * as React from 'react';
-
+import * as ReactDOM from 'react-dom';
+import { FileService } from '../file-service';
 export class LocationListRenderer extends ReactRenderer {
 
     protected _drives: URI[] | undefined;
+    protected doShowTextInput = false;
+    protected lastUniqueTextInputLocation: URI | undefined;
+    protected previousAutocompleteMatch: string;
+    protected doAttemptAutocomplete = true;
 
     constructor(
         protected readonly service: LocationService,
-        host?: HTMLElement
+        host?: HTMLElement,
+        protected readonly fileService?: FileService
     ) {
         super(host);
         this.doLoadDrives();
     }
 
     render(): void {
-        super.render();
+        ReactDOM.render(<React.Fragment>{this.doRender()}</React.Fragment>, this.host, this.doAfterRender);
+    }
+
+    protected doAfterRender = (): void => {
         const locationList = this.locationList;
+        const locationListTextInput = this.locationTextInput;
         if (locationList) {
             const currentLocation = this.service.location;
             locationList.value = currentLocation ? currentLocation.toString() : '';
+        } else if (locationListTextInput) {
+            setTimeout(() => locationListTextInput.focus());
```

Comment:
This could use a comment.

Reply:
Thanks for pointing this out. I did a bit more experimentation and ended up being able to remove the timeout by adding `preventDefault` on the icon's mouseDown event. What was previously happening was the `locationListTextInput.focus()` would fire before the default behavior of the icon's `onClick`, which was causing the document's `activeElement` to switch from the textInput -> toggleIcon, retriggering a blur event and not properly rendering the textInput. 

I've also slightly refactored the event handlers to make each of their roles a bit more clear.